File: Find Total Time Taken[ 15 April 2023]/Solution.py
#  >> PYTHON CODE <<


def totalTime(self,n: int, arr: List[int], time: List[int]) -> int:
        map = {arr[0]: 0}
        ctime = 0
        
        for i in range(1, n):
            lastEncountered = map.get(arr[i], -1)
            
            if lastEncountered == -1:
                ctime += 1
            else:
                if ctime - lastEncountered >= time[arr[i] - 1]:
                    ctime += 1
                else:
                    ctime += time[arr[i] - 1] - (ctime - lastEncountered)
            
            map[arr[i]] = ctime
        
        return ctime

File: Find Total Time Taken[ 15 April 2023]/Solution.java
//  >> JAVA CODE <<


public static long totalTime(int n,int arr[],int time[])
    {
        HashMap<Integer,Long> map=new HashMap<>();
        map.put(arr[0],0L);
        long ctime=0;
        
        for(int i=1;i<n;i++)
        {
            long lastEncountered=map.containsKey(arr[i])?map.get(arr[i]):-1;
            
            if(lastEncountered==-1)
            {
                ctime+=1;
            }
            else
            {
                if((ctime-lastEncountered)>=time[arr[i]-1]) ctime++;
                else ctime+=(time[arr[i]-1]-(ctime-lastEncountered));
            }
            
            map.put(arr[i],ctime);
        }
        
            return ctime;
    }